metagraph/src/annotation/representation/column_compressed/annotate_column_compressed.cpp

```diff
@@ -72,6 +73,42 @@ void ColumnCompressed<Label>::add_labels(const std::vector<Index> &indices,
     }
 }
 
+template <typename Label>
+void ColumnCompressed<Label>::add_label_counts(const std::vector<Index> &indices,
+                                               const VLabels &labels,
+                                               const std::vector<uint32_t> &counts) {
+    assert(indices.size() == counts.size());
+
+    const auto &columns = get_matrix().data();
+
+    if (relation_counts_.size() != columns.size())
+        relation_counts_.resize(columns.size());
+
+    for (const auto &label : labels) {
+        const auto j = label_encoder_.insert_and_encode(label);
+
+        if (!relation_counts_[j].size()) {
+            relation_counts_[j] = sdsl::int_vector<>(columns[j]->num_set_bits(), 0, kCountWidth);
+
+        } else if (relation_counts_[j].size() != columns[j]->num_set_bits()) {
+            logger->error("Binary relation matrix was changed while adding relation counts");
+            exit(1);
+        }
+
+        for (size_t i = 0; i < indices.size(); ++i) {
+            if (uint64_t rank = columns[j]->conditional_rank1(indices[i])) {
+                uint64_t count = std::min(counts[i], (uint32_t)sdsl::bits::lo_set[kCountWidth]);
+                sdsl::int_vector_reference<sdsl::int_vector<>> ref = relation_counts_[j][rank - 1];
+                ref = std::min((uint64_t)ref, sdsl::bits::lo_set[kCountWidth] - count) + count;
+
+            } else {
+                logger->warn("Trying to add count {} for non-annotated object {}",
```

Comment:
Why not say something along the lines of "ignoring count" instead? Saying "trying" makes it sound like there's some attempt to get the operation to work.

Reply:
done